DataDog/datadog-operator
# [datadogfeature] adding orchestrator explorer as a feature

### What does this PR do?

This PR does 2 changes.
- Adding a new `type` the `datadogfeature`, which contains features for the `agent` and `dca`
- Adding orchestrator explorer as a setting for the datadog operator
- Adding the option `processCollection` which defaults to `false` if this is set and `processAgent` is `enabled` live process collection happens as well. 

### Motivation

- Adding orchestrator_explorer to the operator

### Additional Notes

Do we want to add the `DDClusterName` to the process-agent as well? 

### Describe your test plan

- added integration test
- added unit test

manual tests:
- apply supplied `config/samples/datadog-agent-with-clusteragent-with-orchestrator.yaml` with correct API and APP Key settings
either:
- wait for information appear on orchestrator page (e.g. https://dddev.datadoghq.com/orchestration/overview?cols=name%2Cstatus%2Cready%2Crestarts%2Cage%2Clabels&end=1604667730878&paused=false&start=1604666830878&view=pod&tags=kube_cluster_name%3Anam-test) 
- look at DCA and process-agent logs:
process-agent
```
2020-11-06 12:42:32 UTC | PROCESS | INFO | (pkg/forwarder/transaction.go:293 in internalProcess) | Successfully posted payload to "https://orchestrator.datadoghq.com/api/v1/orchestrator", the agent will only log transaction success every 500 transactions
2020-11-06 12:42:32 UTC | PROCESS | INFO | (collector.go:375 in updateStatus) | real time interval updated to 2s
2020-11-06 12:42:42 UTC | PROCESS | INFO | (collector.go:157 in runCheck) | Finished container check #2 in 58.648578ms
2020-11-06 12:42:42 UTC | PROCESS | INFO | (collector.go:157 in runCheck) | Finished pod check #2 in 4.851825ms
```

DCA using a new image containing the updated status information
```
❯ k exec -it orchestrator-cluster-agent-cbd877866-vspfk -- agent status

  Cache Stats
  ===========
    Elements in the cache: 49
    Pods:
      Last Run: (Hits: 1 Miss: 0) | Total: (Hits: 3 Miss: 1)
    Deployments:
      Last Run: (Hits: 10 Miss: 0) | Total: (Hits: 30 Miss: 10)
    ReplicaSets:
      Last Run: (Hits: 23 Miss: 0) | Total: (Hits: 69 Miss: 23)
    Services:
      Last Run: (Hits: 7 Miss: 0) | Total: (Hits: 21 Miss: 7)
    Nodes:
      Last Run: (Hits: 3 Miss: 0) | Total: (Hits: 8 Miss: 4)
```
